admin/commands/pmm/client/upgrade.go

```diff
@@ -0,0 +1,200 @@
+// Copyright 2019 Percona LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package client
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/exec"
+	"strings"
+
+	"github.com/sirupsen/logrus"
+
+	"github.com/percona/pmm/admin/cli/flags"
+	"github.com/percona/pmm/admin/commands"
+	"github.com/percona/pmm/admin/pkg/client/tarball"
+	"github.com/percona/pmm/admin/pkg/common"
+)
+
+// UpgradeCommand is used by Kong for CLI flags and commands.
+type UpgradeCommand struct {
+	Distribution string `enum:"autodetect,package-manager,tarball,docker" default:"autodetect" help:"Type of PMM Client distribution. One of: autodetect,package-manager,tarball,docker"`
+	Version      string `name:"use-version" help:"PMM Client version to upgrade to (default: latest)"`
+
+	InstallPath  string `group:"Tarball flags" default:"/usr/local/percona/pmm2" help:"Path where PMM Client is installed"`
+	User         string `group:"Tarball flags" help:"Set file ownership instead of the current user"`
+	Group        string `group:"Tarball flags" help:"Set group ownership instead of the current group"`
+	SkipChecksum bool   `group:"Tarball flags" help:"Skip checksum validation of the downloaded files"`
+}
+
+type distributionType string
+
+const (
+	distributionAutodetect     distributionType = "autodetect"
+	distributionPackageManager distributionType = "package-manager"
+	distributionTar            distributionType = "tarball"
+	distributionDocker         distributionType = "docker"
+)
+
+type upgradeResult struct{}
+
+// Result is a command run result.
+func (res *upgradeResult) Result() {}
+
+// String stringifies command result.
+func (res *upgradeResult) String() string {
+	return "ok"
```

Comment:
I think message shouldn't be just `ok` :)

Reply:
Agreed. I suggest we keep it like this for now though.
The reason is the UI still needs a lot of polishing. The goal of this PR is to have a functional prototype ready, review it and address the UI in the upcoming PRs.
With this approach I expect the PRs shall be easier to review, not having thousands of lines of changes :)